src/Python/ConfigFile.cpp

```diff
@@ -587,6 +611,123 @@ void ConfigFile::getHadronTriggerLegHistogram(std::string hadronTriggerFile) {
+	file->Close();
+}
+
+void ConfigFile::getNuChiIncorrectPermHistogram(std::string ttbarLikelihoodFile) {
+	using namespace std;
+
+	if (!boost::filesystem::exists(ttbarLikelihoodFile)) {
+		cerr << "ConfigFile::getNuChiIncorrectPermHistogram(" << ttbarLikelihoodFile << "): could not find file" << endl;
+		throw "Could not find file " + ttbarLikelihoodFile;
+	}
+
+	boost::scoped_ptr<TFile> file(TFile::Open(ttbarLikelihoodFile.c_str()));
+	Globals::NuChiIncorrectPermHistogram = (boost::shared_ptr<TH1F>) (TH1F*) file->Get("NeutrinoChiSqFalseB")->Clone();
+	file->Close();
+}
+
+
+void ConfigFile::getHadronicRecoCorrectPermHistogram(std::string ttbarLikelihoodFile) {
+	using namespace std;
+
+	if (!boost::filesystem::exists(ttbarLikelihoodFile)) {
+		cerr << "ConfigFile::getHadronicRecoCorrectPermHistogram(" << ttbarLikelihoodFile << "): could not find file" << endl;
+		throw "Could not find file " + ttbarLikelihoodFile;
+	}
+
+	boost::scoped_ptr<TFile> file(TFile::Open(ttbarLikelihoodFile.c_str()));
+	Globals::HadronicRecoCorrectPermHistogram = (boost::shared_ptr<TH2F>) (TH2F*) file->Get("MT vs MW True Reconstructed B(hadronic)")->Clone();
+	file->Close();
+}
+
+
+void ConfigFile::getHadronicRecoIncorrectPermHistogram(std::string ttbarLikelihoodFile) {
+	using namespace std;
+
+	if (!boost::filesystem::exists(ttbarLikelihoodFile)) {
+		cerr << "ConfigFile::getHadronicRecoIncorrectPermHistogram(" << ttbarLikelihoodFile << "): could not find file" << endl;
+		throw "Could not find file " + ttbarLikelihoodFile;
+	}
+
+	boost::scoped_ptr<TFile> file(TFile::Open(ttbarLikelihoodFile.c_str()));
+	Globals::HadronicRecoIncorrectPermHistogram = (boost::shared_ptr<TH2F>) (TH2F*) file->Get("MT vs MW False Reconstructed B(hadronic)")->Clone();
+	file->Close();
+}
+
+
+void ConfigFile::getLeptonicRecoCorrectPermHistogram(std::string ttbarLikelihoodFile) {
+	using namespace std;
+
+	if (!boost::filesystem::exists(ttbarLikelihoodFile)) {
+		cerr << "ConfigFile::getLeptonicRecoCorrectPermHistogram(" << ttbarLikelihoodFile << "): could not find file" << endl;
+		throw "Could not find file " + ttbarLikelihoodFile;
+	}
+
+	boost::scoped_ptr<TFile> file(TFile::Open(ttbarLikelihoodFile.c_str()));
+	Globals::LeptonicRecoCorrectPermHistogram = (boost::shared_ptr<TH2F>) (TH2F*) file->Get("MT vs MW True Reconstructed B(leptonic)")->Clone();
+	file->Close();
+}
+
+
+void ConfigFile::getLeptonicRecoIncorrectPermHistogram(std::string ttbarLikelihoodFile) {
+	using namespace std;
+
+	if (!boost::filesystem::exists(ttbarLikelihoodFile)) {
+		cerr << "ConfigFile::getLeptonicRecoIncorrectPermHistogram(" << ttbarLikelihoodFile << "): could not find file" << endl;
+		throw "Could not find file " + ttbarLikelihoodFile;
+	}
+
+	boost::scoped_ptr<TFile> file(TFile::Open(ttbarLikelihoodFile.c_str()));
+	Globals::LeptonicRecoIncorrectPermHistogram = (boost::shared_ptr<TH2F>) (TH2F*) file->Get("MT vs MW False Reconstructed B(leptonic)")->Clone();
+	file->Close();
+}
+
+
+
+
+
+
+
```

Comment:
nearing the 'indistinguishable wall of code' number of lines. This need to be shorter and easier to read